### ISSUE TITLE
niriss image3 crashes with "MalformedPolygonError: intersection: find all intersections"

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8513]
_Issue [JP-3636](https://jira.stsci.edu/browse/JP-3636) was created on JIRA by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran):_

while reprocessing with b10.1.2, ops saw nis_image datasets for several programs crash in level 3 processing during the tweakreg step, as documented in DMSOPS-1090. an example is jw01536-o070_20240525t183345_image3_00002 (for product named jw01536-o070_t004_niriss_clearp-f444w-sub128)
```java
2024-05-28 12:52:31,420 - CRDS - DEBUG - Final effective context is 'jwst_1235.pmap'
2024-05-28 12:52:31,420 - CRDS - DEBUG - Computing best references locally.
2024-05-28 12:52:31,421 - CRDS - DEBUG - Bestrefs header:
{'META.EXPOSURE.READPATT [READPATT]': 'NISRAPID',
'META.EXPOSURE.TYPE [EXP_TYPE]': 'NIS_IMAGE',
'META.INSTRUMENT.DETECTOR [DETECTOR]': 'NIS',
'META.INSTRUMENT.FILTER [FILTER]': 'F444W',
'META.INSTRUMENT.LAMP_STATE [LAMP]': 'NONE',
'META.INSTRUMENT.NAME [INSTRUME]': 'NIRISS',
'META.INSTRUMENT.PUPIL [PUPIL]': 'CLEARP',
'META.OBSERVATION.DATE [DATE-OBS]': '2022-08-09',
'META.OBSERVATION.TIME [TIME-OBS]': '13:43:41.131',
'META.SUBARRAY.NAME [SUBARRAY]': 'SUB128',
'META.VISIT.CROWDED_FIELD [CROWDFLD]': 'F',
'META.VISIT.TSOVISIT [TSOVISIT]': 'F',
'META.VISIT.TYPE [VISITYPE]': 'PRIME_TARGETED_FIXED',
'REFTYPE': 'UNDEFINED'}
2024-05-28 12:52:31,422 - CRDS - DEBUG - Reference type 'abvegaoffset' defined as 'jwst_niriss_abvegaoffset_0003.asdf'
2024-05-28 12:52:31,422 - CRDS - DEBUG - Reference type 'apcorr' defined as 'jwst_niriss_apcorr_0008.fits'
2024-05-28 12:52:31,422 - CRDS - DEBUG - Reference type 'drizpars' defined as 'jwst_niriss_drizpars_0002.fits'
2024-05-28 12:52:31,438 - stpipe.Image3Pipeline - INFO - Prefetch for ABVEGAOFFSET reference file is '/ifs/archive/ops/jwst/ref/tmp_crds/crds/cache/references/jwst/niriss/jwst_niriss_abvegaoffset_0003.asdf'.
2024-05-28 12:52:31,439 - stpipe.Image3Pipeline - INFO - Prefetch for APCORR reference file is '/ifs/archive/ops/jwst/ref/tmp_crds/crds/cache/references/jwst/niriss/jwst_niriss_apcorr_0008.fits'.
2024-05-28 12:52:31,440 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/ifs/archive/ops/jwst/ref/tmp_crds/crds/cache/references/jwst/niriss/jwst_niriss_drizpars_0002.fits'.
2024-05-28 12:52:31,442 - stpipe.Image3Pipeline - INFO - Starting calwebb_image3 ...
2024-05-28 12:52:32,252 - stpipe.Image3Pipeline.tweakreg - INFO - Step tweakreg running with args (<ModelContainer>,).
2024-05-28 12:52:32,253 - stpipe.Image3Pipeline.tweakreg - INFO - Step tweakreg parameters are: {'pre_hooks': [], 'post_hooks': [], 'output_file': None, 'output_dir': None, 'output_ext': '.fits', 'output_use_model': True, 'output_use_index': True, 'save_results': False, 'skip': False, 'suffix': None, 'search_output_file': True, 'input_dir': '/ifs/archive/ops/jwst/info/owlmgr/paths/sdp/asn_creation/cal/level3', 'save_catalogs': False, 'use_custom_catalogs': False, 'catalog_format': 'ecsv', 'catfile': '', 'kernel_fwhm': 2.5, 'snr_threshold': 10, 'sharplo': 0.2, 'sharphi': 1.0, 'roundlo': -1.0, 'roundhi': 1.0, 'brightest': 100, 'peakmax': None, 'bkg_boxsize': 400, 'enforce_user_order': False, 'expand_refcat': False, 'minobj': 15, 'searchrad': 1.0, 'use2dhist': True, 'separation': 1.0, 'tolerance': 1.0, 'xoffset': 0.0, 'yoffset': 0.0, 'fitgeometry': 'shift', 'nclip': 3, 'sigma': 3.0, 'abs_refcat': 'GAIADR3', 'save_abs_catalog': False, 'abs_minobj': 15, 'abs_searchrad': 6.0, 'abs_use2dhist': True, 'abs_separation': 0.1, 'abs_tolerance': 0.7, 'abs_fitgeometry': 'rshift', 'abs_nclip': 3, 'abs_sigma': 3.0}
2024-05-28 12:52:32,277 - stpipe.Image3Pipeline.tweakreg - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/photutils/background/background_2d.py:273: AstropyUserWarning: Input data contains invalid values (NaNs or infs), which were automatically masked.
2024-05-28 12:52:32,277 - stpipe.Image3Pipeline.tweakreg - WARNING - warnings.warn('Input data contains invalid values (NaNs or '
2024-05-28 12:52:32,277 - stpipe.Image3Pipeline.tweakreg - WARNING -
2024-05-28 12:52:32,279 - stpipe.Image3Pipeline.tweakreg - INFO - Background could not be estimated in meshes. Using the entire unmasked array for background estimation: bkg_boxsize=(128, 128).
2024-05-28 12:52:32,289 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 3 sources in jw01536070001_0210l_00001_nis_cal.fits.
2024-05-28 12:52:32,317 - stpipe.Image3Pipeline.tweakreg - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/photutils/background/background_2d.py:273: AstropyUserWarning: Input data contains invalid values (NaNs or infs), which were automatically masked.
2024-05-28 12:52:32,317 - stpipe.Image3Pipeline.tweakreg - WARNING - warnings.warn('Input data contains invalid values (NaNs or '
2024-05-28 12:52:32,317 - stpipe.Image3Pipeline.tweakreg - WARNING -
2024-05-28 12:52:32,318 - stpipe.Image3Pipeline.tweakreg - INFO - Background could not be estimated in meshes. Using the entire unmasked array for background estimation: bkg_boxsize=(128, 128).
2024-05-28 12:52:32,326 - stpipe.Image3Pipeline.tweakreg - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/photutils/detection/daofinder.py:673: NoDetectionsWarning: Sources were found, but none pass the sharpness, roundness, or peakmax criteria
2024-05-28 12:52:32,326 - stpipe.Image3Pipeline.tweakreg - WARNING - warnings.warn('Sources were found, but none pass the sharpness, '
2024-05-28 12:52:32,326 - stpipe.Image3Pipeline.tweakreg - WARNING -
2024-05-28 12:52:32,327 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw01536070001_0210l_00002_nis_cal.fits.
2024-05-28 12:52:32,339 - stpipe.Image3Pipeline.tweakreg - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/photutils/background/background_2d.py:273: AstropyUserWarning: Input data contains invalid values (NaNs or infs), which were automatically masked.
2024-05-28 12:52:32,340 - stpipe.Image3Pipeline.tweakreg - WARNING - warnings.warn('Input data contains invalid values (NaNs or '
2024-05-28 12:52:32,340 - stpipe.Image3Pipeline.tweakreg - WARNING -
2024-05-28 12:52:32,341 - stpipe.Image3Pipeline.tweakreg - INFO - Background could not be estimated in meshes. Using the entire unmasked array for background estimation: bkg_boxsize=(128, 128).
2024-05-28 12:52:32,348 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 2 sources in jw01536070001_0210l_00003_nis_cal.fits.
2024-05-28 12:52:32,360 - stpipe.Image3Pipeline.tweakreg - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/photutils/background/background_2d.py:273: AstropyUserWarning: Input data contains invalid values (NaNs or infs), which were automatically masked.
2024-05-28 12:52:32,360 - stpipe.Image3Pipeline.tweakreg - WARNING - warnings.warn('Input data contains invalid values (NaNs or '
2024-05-28 12:52:32,360 - stpipe.Image3Pipeline.tweakreg - WARNING -
2024-05-28 12:52:32,362 - stpipe.Image3Pipeline.tweakreg - INFO - Background could not be estimated in meshes. Using the entire unmasked array for background estimation: bkg_boxsize=(128, 128).
2024-05-28 12:52:32,370 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 1 sources in jw01536070001_0210l_00004_nis_cal.fits.
2024-05-28 12:52:32,381 - stpipe.Image3Pipeline.tweakreg - INFO -
2024-05-28 12:52:32,382 - stpipe.Image3Pipeline.tweakreg - INFO - Number of image groups to be aligned: 4.
2024-05-28 12:52:32,382 - stpipe.Image3Pipeline.tweakreg - INFO - Image groups:
2024-05-28 12:52:32,407 - stpipe.Image3Pipeline.tweakreg - INFO - * Images in GROUP 'jw01536070001_0210l_00001_nis_cal':
2024-05-28 12:52:32,407 - stpipe.Image3Pipeline.tweakreg - INFO - jw01536070001_0210l_00001_nis_cal
2024-05-28 12:52:32,434 - stpipe.Image3Pipeline.tweakreg - INFO - * Images in GROUP 'jw01536070001_0210l_00002_nis_cal':
2024-05-28 12:52:32,434 - stpipe.Image3Pipeline.tweakreg - INFO - jw01536070001_0210l_00002_nis_cal
2024-05-28 12:52:32,460 - stpipe.Image3Pipeline.tweakreg - INFO - * Images in GROUP 'jw01536070001_0210l_00003_nis_cal':
2024-05-28 12:52:32,460 - stpipe.Image3Pipeline.tweakreg - INFO - jw01536070001_0210l_00003_nis_cal
2024-05-28 12:52:32,486 - stpipe.Image3Pipeline.tweakreg - INFO - * Images in GROUP 'jw01536070001_0210l_00004_nis_cal':
2024-05-28 12:52:32,486 - stpipe.Image3Pipeline.tweakreg - INFO - jw01536070001_0210l_00004_nis_cal
2024-05-28 12:52:32,486 - stpipe.Image3Pipeline.tweakreg - INFO -
2024-05-28 12:52:32,487 - stpipe.Image3Pipeline.tweakreg - INFO -
2024-05-28 12:52:32,487 - stpipe.Image3Pipeline.tweakreg - INFO - ***** tweakwcs.imalign.align_wcs() started on 2024-05-28 12:52:32.487017
2024-05-28 12:52:32,487 - stpipe.Image3Pipeline.tweakreg - INFO - Version 0.8.5
2024-05-28 12:52:32,487 - stpipe.Image3Pipeline.tweakreg - INFO -
2024-05-28 12:52:32,553 - stpipe.Image3Pipeline.tweakreg - WARNING - Group with ID 'jw01536070001_0210l_00002_nis_cal' will not be aligned: empty source catalog
2024-05-28 12:52:32,881 - stpipe.Image3Pipeline.tweakreg - INFO - Selected image 'GROUP ID: jw01536070001_0210l_00003_nis_cal' as reference image
2024-05-28 12:52:32,886 - stpipe.Image3Pipeline.tweakreg - INFO - Aligning image catalog 'GROUP ID: jw01536070001_0210l_00004_nis_cal' to the reference catalog.
2024-05-28 12:52:32,959 - stpipe.Image3Pipeline.tweakreg - INFO - Matching sources from 'jw01536070001_0210l_00004_nis_cal' catalog with sources from the reference 'jw01536070001_0210l_00003_nis_cal' catalog.
2024-05-28 12:52:32,960 - stpipe.Image3Pipeline.tweakreg - INFO - Computing initial guess for X and Y shifts...
2024-05-28 12:52:32,961 - stpipe.Image3Pipeline.tweakreg - WARNING - Unable to estimate significance of the detection of the initial shift.
2024-05-28 12:52:32,961 - stpipe.Image3Pipeline.tweakreg - INFO - Found initial X and Y shifts of 0.2428, -0.08424 (arcsec) with 1 matches.
2024-05-28 12:52:32,962 - stpipe.Image3Pipeline.tweakreg - INFO - Found 1 matches for 'GROUP ID: jw01536070001_0210l_00004_nis_cal'...
2024-05-28 12:52:32,962 - stpipe.Image3Pipeline.tweakreg - WARNING - Not enough matches (< 1) found for image catalog 'GROUP ID: jw01536070001_0210l_00004_nis_cal'.
2024-05-28 12:52:32,974 - stpipe.Image3Pipeline.tweakreg - INFO - Added 0 unmatched sources from 'GROUP ID: jw01536070001_0210l_00004_nis_cal' to the reference catalog.
Traceback (most recent call last):
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/stpipe/cli/strun.py", line 24, in main
Step.from_cmdline(sys.argv[1:])
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 187, in from_cmdline
return cmdline.step_from_cmdline(args)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/stpipe/cmdline.py", line 392, in step_from_cmdline
step.run(*positional)
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 479, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_image3.py", line 89, in process
input_models = self.tweakreg(input_models)
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 479, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/jwst/tweakreg/tweakreg_step.py", line 308, in process
align_wcs(
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/tweakwcs/imalign.py", line 705, in align_wcs
current_wcat = max_overlap_image(
^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/tweakwcs/imalign.py", line 870, in max_overlap_image
idx = np.argmax([refimage.intersection_area(im) for im in images])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/tweakwcs/imalign.py", line 870, in <listcomp>
idx = np.argmax([refimage.intersection_area(im) for im in images])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/tweakwcs/wcsimage.py", line 1544, in intersection_area
self.polygon.intersection(wim.polygon).area()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/spherical_geometry/polygon.py", line 1241, in intersection
subpolygons = polya.intersection(polyb)
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/spherical_geometry/polygon.py", line 660, in intersection
return g.intersection()
^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/spherical_geometry/graph.py", line 437, in intersection
self._sanity_check("intersection: find all intersections")
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/spherical_geometry/graph.py", line 389, in _sanity_check
raise MalformedPolygonError(msg)
spherical_geometry.polygon.MalformedPolygonError: intersection: find all intersections
2024149125234 INFO src=ssb_calibration_wrapper._strun_analyze_and_log_failure fsn=jw01536-o070_20240525t183345_image3_00002_asn msg="strun /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.13.3.20240108-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_image3.cfg FAILED (exit=1) on jw01536-o070_20240525t183345_image3_00002_asn.json."
2024149125234 ERROR src=ssb_calibration_wrapper.calibrate._strun._strun_analyze_and_log_failure fsn=jw01536-o070_20240525t183345_image3_00002_asn msg="strun exit status=1" ```
the same dataset made it through level 3 successfully prior to b10.1 without error.  for example, 

jw01536-o070_20231105t051444_image3_00002 **jw01536-o070_t004_niriss_clearp-f444w-sub128 < ok b9.3.1**

jw01536-o070_20240328t063536_image3_00002 **jw01536-o070_t004_niriss_clearp-f444w-sub128 < failed b10.1**

jw01536-o070_20240525t183345_image3_00002 **jw01536-o070_t004_niriss_clearp-f444w-sub128 < failed  b10.1**

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=701243#comment-701243):_

Very bad looking step parameter values being assigned by the active pars-tweakregstep ref file for NIRISS images. Wouldn't be surprised if that's what's causing the failures.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=701538#comment-701538):_

[Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran) Could you make the image3 asn file and the cal files listed therein available somewhere for SCSB to do testing? We want to make sure to use the exact same inputs as used in Ops.

**Comment by stscijgbot-jp** [comment]
_Comment by [Rachel Plesha](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rplesha) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=701647#comment-701647):_

[Kevin Volk](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kvolk), [Jo Taylor](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jotaylor), [André Martel](https://jira.stsci.edu/secure/ViewProfile.jspa?name=martel), [Tony Sohn](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tsohn), [Paul Goudfrooij](https://jira.stsci.edu/secure/ViewProfile.jspa?name=goudfroo) , I wanted to bring this ticket to your attention. There were some thoughts at the JP meeting yesterday that this might be due to some parameters that are set to bad values, specifically in the search radius and tolerance values when using tweakreg with the gaia3 absolute alignment in the parameter reference file. I'm not sure if any of this will be affected or fixed by the upcoming changes using the IRAF starfinder, but please chime in if you have any ideas about this error or upcoming updates to the values.

**Comment by stscijgbot-jp** [comment]
_Comment by [Kevin Volk](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kvolk) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=701651#comment-701651):_

The issue here seems to me to be that the image size is too small and there are not enough stars to make the tweakreg catalogue.  This example is a photometric calibration observation with image size 128x128 pixels.  The image attached shows the field in the rate image.  There is just one star, the photometric standard, present.  Apparently tweakreg is not finding this star (which is a problem in itself given the image...) and produces no catalogues for matching and so things crash.

The question is why it is crashing currently as opposed to in previous runs.  The revisions to the tweakreg parameter files should not cause problems in an image like this.  Nonetheless that may be the issue.

**Comment by stscijgbot-jp** [comment]
_Comment by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=701856#comment-701856):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) i put the data in ████████████████████████████

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=702518#comment-702518):_

Are there specific instructions that could help me reproduce this issue?

I run:
```java
pip install -r reqs_macos-stable-deps.txt```
where reqs_macos-stable-deps.txt was downloaded from here: <https://ssb.stsci.edu/releases/jwstdp/1.13.3/reqs_macos-stable-deps.txt>

I then run the step using strun, creating TweakRegStep object or creating Image3Pipeline object and running the step. Each time the step finished successfully (success here means not crashing, not performing a successful alignment).

**Comment by stscijgbot-jp** [comment]
_Comment by [John Scott](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jscott) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=704534#comment-704534):_

There is a possibly related error in [jw01537-o003_20240616t141518_image3_00004](https://pljwdmsweb.stsci.edu/owlgui/logs/directory_view?directory_path=/ifs/archive/ops/jwst/info/owl/logs/owlmgr_jw01537-o003_20240616t141518_image3_00004_1718556837.779796&fileset_name=jw01537-o003_20240616t141518_image3_00004&dag_name=REPRO_LEVEL_3): Same part of the code, and other products in this observations have the malformed polygon error...
```java
2024-06-16 17:01:43,594 - stpipe.Image3Pipeline.tweakreg - INFO - ***** tweakwcs.imalign.align_wcs() started on 2024-06-16 17:01:43.5945952024-06-16 17:01:43,594 - stpipe.Image3Pipeline.tweakreg - INFO - Version 0.8.62024-06-16 17:01:43,594 - stpipe.Image3Pipeline.tweakreg - INFO - 2024-06-16 17:01:44,551 - stpipe.Image3Pipeline.tweakreg - WARNING - Group with ID '987654' will not be aligned: empty source catalogTraceback (most recent call last):File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/stpipe/cli/strun.py", line 24, in mainStep.from_cmdline(sys.argv[1:])File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 191, in from_cmdlinereturn cmdline.step_from_cmdline(args)^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/stpipe/cmdline.py", line 394, in step_from_cmdlinestep.run(*positional)File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 509, in runstep_result = self.process(*args)^^^^^^^^^^^^^^^^^^^File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_image3.py", line 89, in processinput_models = self.tweakreg(input_models)^^^^^^^^^^^^^^^^^^^^^^^^^^^File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 509, in runstep_result = self.process(*args)^^^^^^^^^^^^^^^^^^^File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/jwst/tweakreg/tweakreg_step.py", line 476, in processalign_wcs(File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.14.0.20240401-py3.11/lib/python3.11/site-packages/tweakwcs/imalign.py", line 652, in align_wcsraise ValueError("Too few input images (or groups of images) with "ValueError: Too few input images (or groups of images) with non-empty catalogs. ```
ValueError: Too few input images (or groups of images) with non-empty catalogs

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=709356#comment-709356):_

should be fixed via spacetelescope/tweakwcs#205 and tweakwcs release 0.8.8 and #8657

**Comment by stscijgbot-jp** [comment]
_Comment by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran) on [JIRA](https://jira.stsci.edu/browse/JP-3636?focusedCommentId=944991#comment-944991):_

don't see this type of failures in our latest repro with b11.1.1. considered fixed, closing,
